networkx/algorithms/centrality/domirank.py

```diff
@@ -0,0 +1,305 @@
+"""DomiRank Centrality: By Marcus Engsig (@mengsig)"""
+
+import networkx as nx
+from networkx.utils import not_implemented_for
+
+__all__ = ["domirank"]
+
+
+@not_implemented_for("multigraph")
+@nx._dispatchable(edge_attrs="weight")
+def domirank(
+    G, analytical=False, sigma=0.95, dt=0.1, epsilon=1e-5, max_iter=1000, check_step=10
+):
+    r"""Compute the DomiRank centrality for the graph G.
+
+    DomiRank centrality computes the centrality for a node by adding
+    1 minus the centrality of its neighborhood. This essentially finds the
+    dominance of a node in its neighborhood, where the parameter $\sigma$ determines
+    the amount of competition in the system. The competition parameter $\sigma$
+    tunes the balance of DomiRanks integration of  local and global topological
+    information, to find nodes that are either locally or globally Dominant. It is
+    important to note that for the iterative formulation of DomiRank (as seen below),
+    the competition parameter is bounded: $\sigma \in [0,\frac{1}{-lambda_N}].
```

Comment:
```suggestion
    the competition parameter is bounded: $\sigma \in [0,1/(-\lambda_N)]$.
```

Where does `\lambda_N` come from? I'm assuming this refers to eigenvalues and is explained in the companion paper, but I think docstrings should be self-contained.

I didn't read the paper, so I might be asking a dumb question here: why does  this mention `[0, 1/\lambda_N]` whereas the code uses `[0, 1]`?

Reply:
Re: my second paragraph, I see now that you're dividing by the smallest eigenvalue before checking the value of sigma. The exception messages will have to be changed, then.